pbio_test.go

```diff
@@ -0,0 +1,82 @@
+package pbio_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/bsm/pbio"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var message, _ = structpb.NewValue(map[string]interface{}{
+	"firstName": "John",
+	"lastName":  "Smith",
+	"isAlive":   true,
+	"age":       27,
+	"address": map[string]interface{}{
+		"streetAddress": "21 2nd Street",
+		"city":          "New York",
+		"state":         "NY",
+		"postalCode":    "10021-3100",
+	},
+})
+
+var _ = Describe("Encoder", func() {
+	var subject *pbio.Encoder
+	var buf *bytes.Buffer
+
+	BeforeEach(func() {
+		buf = new(bytes.Buffer)
+		subject = pbio.NewEncoder(buf)
+	})
+
+	It("should encode", func() {
+		Expect(subject.Encode(message)).To(Succeed())
+		Expect(buf.Len()).To(BeNumerically("~", 192, 20))
+		Expect(subject.Encode(message)).To(Succeed())
+		Expect(buf.Len()).To(BeNumerically("~", 384, 20))
```

Comment:
This approach to testing marshalling is dangerous, IMO.

Maybe it's an excess test - the one you did for Decoder is what gogo does + it tests everything both marshalling and unmarshalling.